src/array.rs

```diff
@@ -21,6 +21,21 @@ impl<T: RuntimeType> Array<T> {
         Self::default()
     }
 
+    pub fn with_len(len: usize) -> Self {
+        unsafe {
+            assert!(len < 0xFFFF_FFFF as usize);
+            let data = CoTaskMemAlloc(len * std::mem::size_of::<T>()) as *mut T;
+
+            if data.is_null() {
+                panic!("Could not successfully allocate for Array");
+            }
+
+            std::ptr::write_bytes(data, 0, len);
```

Comment:
In the spirt of better documentation around safety, can we document this is safe because `RuntimeTypes` are safe to zero-initialize?